### PR TITLE
Task #8288: Handle Event for User Setting Screen

### PR DESCRIPTION
**Pull request by tanbtd-1747**
### Issue Link :link:
https://edu-redmine.sun-asterisk.vn/issues/8288

### Goals :soccer:
- Select Wallets Management row to go to the Wallets Management Screen
- Switch On/Off Slider to turn on or off the notification. If the Notification is turned on, select the TimePicker to set the notification time
- Select Change password row to update password
- Select Logout row to deauthorize account and go to the Log in Screen

### CODE REVIEW

**Comment by tanbtd-1747** [comment]
ready

File: Money*/Money*/Scenes/UserSetting/UserSettingViewController.swift [hunk 1]
```diff
@@ -7,22 +7,230 @@
...
     // MARK: - IBOutlets
     @IBOutlet private var notificationTimeLabel: UILabel!
+    @IBOutlet private var notificationSwitch: UISwitch!
     
     // MARK: - Private properties
     var user: User!
+    var notificationTime: Date?
+    var isNotificationEnabled: Bool?
+    var timePicker: UIDatePicker?
+    var isTimePickerHidden = true
```
Comment (tgdong2296) [inline Money*/Money*/Scenes/UserSetting/UserSettingViewController.swift]: phía trên ghi là `// MARK: - Private properties` mà dưới không để private à anh :v

File: Money*/Money*/Scenes/UserSetting/UserSettingViewController.swift [hunk 2]
```diff
@@ -7,22 +7,230 @@
...
+            notificationTimeLabel.text = formatter.string(from: notificationTime)
+        }
+    }
+    
+    private func loadUserSetting() {
+        isNotificationEnabled = UserDefaults.standard.value(forKey: Identifier.keyIsNofiticationEnabled) as? Bool
+        notificationTime = UserDefaults.standard.value(forKey: Identifier.keyNotificationTime) as? Date
+    }
+    
+    private func registerUserNotification() {
```
Comment (tgdong2296) [inline Money*/Money*/Scenes/UserSetting/UserSettingViewController.swift]: register notification để trong `AppDelegate` anh nhé

File: Money*/Money*/Scenes/UserSetting/UserSettingViewController.swift [hunk 3]
```diff
@@ -7,22 +7,230 @@
...
+        dateComponents.hour = Calendar.current.component(.hour, from: time)
+        dateComponents.minute = Calendar.current.component(.minute, from: time)
+        
+        let trigger = UNCalendarNotificationTrigger(dateMatching: dateComponents, repeats: true)
+
+        let request = UNNotificationRequest(identifier: UUID().uuidString, content: content, trigger: trigger)
+        notificationCenter.add(request, withCompletionHandler: nil)
+    }
+    
+    private func unscheduleUserNotification() {
```
Comment (tgdong2296) [inline Money*/Money*/Scenes/UserSetting/UserSettingViewController.swift]: tương tự, những hàm xử lý notification nên tách ra 1 class quản trị riêng và sử dụng singleton cho nó

File: Money*/Money*/Scenes/UserSetting/UserSettingViewController.swift [hunk 4]
```diff
@@ -7,22 +7,230 @@
...
+        let notificationCenter = UNUserNotificationCenter.current()
+        notificationCenter.removeAllPendingNotificationRequests()
+    }
+    
+    private func showHideTimePicker() {
+        if isTimePickerHidden {
+            timePicker = UIDatePicker()
```
Comment (tgdong2296) [inline Money*/Money*/Scenes/UserSetting/UserSettingViewController.swift]: nên khởi tạo luôn khi khai báo

**Comment by tanbtd-1747** [comment]
ready

**Comment by tienpx3010** [comment]
merged

**Review by tgdong2296 — APPROVED** [review]

File: Money*/Money*/Data/DailyNotification.swift [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    var isNotificationEnabled = false
+    
+    static let shared = DailyNotification()
+    
+    init() {
+        if let time = UserDefaults.standard.value(forKey: Identifier.keyNotificationTime) as? Date {
+            notificationTime = time
+        } else {
+            var dateComponents = DateComponents()
+            dateComponents.hour = 22
```
Comment (tienpx3010) [inline Money*/Money*/Data/DailyNotification.swift:23]: Set contants default

**Review by tienpx3010 — APPROVED** [review]

**Review by tienpx3010 — APPROVED** [review]

**Pull request merged**